### PR TITLE
ci(lint): forbid wrong typing inside the CI

### PR DESCRIPTION
**Pull request by Sigurd-Borge**
Current tox.ini and ci.yml doesn't actually fail if files are incorrectly written (it tries to fix with `ruff check --fix`) or formatted according to the rules in ruff.toml, this corrects that.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -27,3 +26,10 @@ commands =
     ruff format src/ tests/ {posargs}
     mypy {posargs}
 
+[testenv:lint-github]
+description = Linting and formatting with ruff, typing with mypy
```
Comment (MartinBelthle) [inline tox.ini:30]: You should add the licenser checker with --action=check-strict

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,7 +29,9 @@ jobs:
 
       - name: Performs Ubuntu tests
         if: matrix.os != 'windows-latest'
-        run: tox -p
+        run: |
+          tox -e lint-github
```
Comment (MartinBelthle) [inline .github/workflows/ci.yml:33]: Change this to lint-ci ?

**Review by MartinBelthle — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (Sigurd-Borge) [reply tox.ini:30]: Like minds, already done ;)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Sigurd-Borge) [reply .github/workflows/ci.yml:33]: If you want :)

**Review by MartinBelthle — APPROVED** [review]

**Pull request merged**